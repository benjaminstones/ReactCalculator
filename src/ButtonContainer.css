.button-container {
    background-color: #202124;
    display: flex;
    flex-direction: row;
    flex-wrap: wrap;
    flex: 1 0 auto;
  }

  .button-container > div {
    width: 100%;
    margin-bottom: 1px;
    flex: 1 0 auto;
    display: flex;
  }

  .button-row {
    height: 70px;
  }