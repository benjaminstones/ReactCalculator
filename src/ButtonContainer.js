import './ButtonContainer.css'
import Button from './Button';

function ButtonContainer(props) {

    const handleClick = (buttonName, buttonType) => {
        props.clickHandler(buttonName, buttonType);
      };

    return (
      <div className='button-container'>        
        <div className='button-row'> 
          <Button btnName={'AC'} btnType={'util'} clickHandler={handleClick}/>
          <Button btnName={'('} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={')'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'del'} btnType={'util'} clickHandler={handleClick}/>
        </div>
        <div className='button-row'> 
          <Button btnName={'1'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'2'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'3'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'+'} btnType={'op'} clickHandler={handleClick}/>
        </div>
        <div className='button-row'> 
          <Button btnName={'4'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'5'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'6'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'-'} btnType={'op'} clickHandler={handleClick}/>

        </div>
        <div className='button-row'> 
          <Button btnName={'7'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'8'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'9'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'*'} btnType={'op'} clickHandler={handleClick}/>
        </div>
        <div className='button-row'> 
          <Button btnName={'.'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'0'} btnType={'num'} clickHandler={handleClick}/>
          <Button btnName={'='} btnType={'util'} clickHandler={handleClick}/>
          <Button btnName={'/'} btnType={'op'} clickHandler={handleClick}/>
        </div>

      </div>
    );
  }
  
  export default ButtonContainer;