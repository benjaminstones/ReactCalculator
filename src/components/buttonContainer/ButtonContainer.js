import { NON_OPERATOR_CHAR, OPERATOR_CHAR, UTIL } from '../../Constants.js';
import Button from '../button/Button';
import './ButtonContainer.css';

function ButtonContainer(props) {

	const handleClick = (buttonName, buttonType) => {
		props.clickHandler(buttonName, buttonType);
	};

	return (
		<div className='button-container'>
			<div className='button-row'>
				<Button btnName={'AC'} btnType={UTIL} clickHandler={handleClick} />
				<Button btnName={'('} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={')'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'del'} btnType={UTIL} clickHandler={handleClick} />
			</div>
			<div className='button-row'>
				<Button btnName={'1'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'2'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'3'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'+'} btnType={OPERATOR_CHAR} clickHandler={handleClick} />
			</div>
			<div className='button-row'>
				<Button btnName={'4'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'5'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'6'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'-'} btnType={OPERATOR_CHAR} clickHandler={handleClick} />

			</div>
			<div className='button-row'>
				<Button btnName={'7'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'8'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'9'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'\u00d7'} btnType={OPERATOR_CHAR} clickHandler={handleClick} />
			</div>
			<div className='button-row'>
				<Button btnName={'.'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'0'} btnType={NON_OPERATOR_CHAR} clickHandler={handleClick} />
				<Button btnName={'='} btnType={UTIL} clickHandler={handleClick} />
				<Button btnName={'\u00f7'} btnType={OPERATOR_CHAR} clickHandler={handleClick} />
			</div>

		</div>
	);
}

export default ButtonContainer;