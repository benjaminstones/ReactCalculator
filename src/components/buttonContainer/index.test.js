import React from 'react';
import { shallow } from 'enzyme';
import '../../setupTests'
import ButtonContainer from './index.js';

let wrapped = shallow(<ButtonContainer/>);
describe('ButtonContainer', () => {
	it('should render the ButtonContainer Component correctly', () => {   
		expect(wrapped).toMatchSnapshot();
	});
});