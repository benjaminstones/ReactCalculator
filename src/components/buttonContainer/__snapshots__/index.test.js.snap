// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonContainer should render the ButtonContainer Component correctly 1`] = `
<styled.div>
  <styled.div
    className="button-row"
  >
    <Button
      btnName="AC"
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "UTIL": "UTIL",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="("
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName=")"
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="del"
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "UTIL": "UTIL",
        }
      }
      clickHandler={[Function]}
    />
  </styled.div>
  <styled.div
    className="button-row"
  >
    <Button
      btnName="1"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="2"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="3"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="+"
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "OPERATOR_CHAR": "OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
  </styled.div>
  <styled.div
    className="button-row"
  >
    <Button
      btnName="4"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="5"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="6"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="-"
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "OPERATOR_CHAR": "OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
  </styled.div>
  <styled.div
    className="button-row"
  >
    <Button
      btnName="7"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="8"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="9"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="ร"
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "OPERATOR_CHAR": "OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
  </styled.div>
  <styled.div
    className="button-row"
  >
    <Button
      btnName="."
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="0"
      btnTags={
        Object {
          "NON_OPERATOR_CHAR": "NON_OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="="
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "UTIL": "UTIL",
        }
      }
      clickHandler={[Function]}
    />
    <Button
      btnName="รท"
      btnTags={
        Object {
          "HIGHLIGHTED": "HIGHLIGHTED",
          "OPERATOR_CHAR": "OPERATOR_CHAR",
        }
      }
      clickHandler={[Function]}
    />
  </styled.div>
</styled.div>
`;
